apis/v1alpha1/httproute_types.go

```diff
@@ -326,6 +345,37 @@ type HTTPHeaderMatch struct {
 	Values map[string]string `json:"values"`
 }
 
+// HTTPQueryParamMatch describes how to select a HTTP route by matching HTTP
+// query parameters.
+type HTTPQueryParamMatch struct {
+	// Type specifies how to match against the value of the query parameter.
+	//
+	// Support: Extended (Exact, Presence)
+	//
+	// Support: Custom (RegularExpression, ImplementationSpecific)
+	//
+	// Since RegularExpression PathType has custom conformance, implementations
+	// can support POSIX, PCRE or any other dialects of regular expressions.
+	// Please read the implementation's documentation to determine the supported
+	// dialect.
+	//
+	// HTTP query parameter matching MUST be case-sensitive.
```

Comment:
Both key and value? If yes, let's be precise.

Reply:
Agreed, clarified that both key and value should be case-sensitive. This is based on the following guidance from the RFC (emphasis mine):

The scheme and host are case-insensitive and normally provided in lowercase; **all other components are compared in a case-sensitive manner**.

https://tools.ietf.org/html/rfc7230#page-19

Reply:
Case-sensitive is a source of confusion and conflict usually, although I'm not sure how much of that is a problem with query param matching.
Can we include the link to the RFC and mention the section which states the case-sensitiveness?

Reply:
Added in #636